ovc/machines.go

```diff
@@ -62,6 +62,7 @@ type EmptyMachineConfig struct {
 	Description  string `json:"description,omitempty"`
 	Memory       int    `json:"memory,omitempty"`
 	Vcpus        int    `json:"vcpus,omitempty"`
+	Imagetype    string `json:"imagetype,omitempty"`
```

Comment:
The permitted values should be documented.

Taking it a step further, since this adds a new mandatory parameter to the call (= old code is broken by it): I suggest to seize the opportunity and start improving the API by preventing users from creating an (invalid)`EmptyMachineConfig` and instead forcing them to go through a constructor function:
```func NewEmptyMachineConfig(/* mandatory args */) emptyMachineConfig```
.
If the set of permitted image types is closed (@grimpy ?), it could be passed to that constructor in an enum-like style instead of as a free form string:

```
type ImageType int
const (
   ImageTypeLinux ImageType = iota
   ImageTypeWindows
   ....
)
```

Reply:
`golint` throws a spanner in the works of the forced constructor idea above (unexported types " can be annoying to use").
To be continued in #39